hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/StorageDistributionEndpoint.java

```diff
@@ -0,0 +1,199 @@
+    long totalPendingAtOmSide = calculatePendingSizes();
+    long totalOpenKeySize = calculateOpenKeySizes();
+    long totalCommittedSize = calculateCommittedSize();
+    long totalUsedNamespace = totalPendingAtOmSide + totalOpenKeySize + totalCommittedSize;
+    metrics.put("totalPendingAtOmSide", totalPendingAtOmSide);
+    metrics.put("totalOpenKeySize", totalOpenKeySize);
+    metrics.put("totalCommittedSize", totalCommittedSize);
+    metrics.put("totalUsedNamespace", totalUsedNamespace);
+    return metrics;
+  }
+
+  private StorageCapacityDistributionResponse buildStorageDistributionResponse(
+      List<DatanodeStorageReport> nodeStorageReports,
+      GlobalStorageReport storageMetrics,
+      Map<String, Long> namespaceMetrics) {
+    DeletedBlocksTransactionSummary scmSummary;
+    try {
+      scmSummary = scmClient.getDeletedBlockSummary();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+    long totalPendingAtDnSide = nodeStorageReports.stream().mapToLong(DatanodeStorageReport::getPendingDeletions).sum();
+
+    DeletionPendingBytesByStage deletionPendingBytesByStage =
+        createDeletionPendingBytesByStage(namespaceMetrics.get("totalPendingAtOmSide"),
+            scmSummary.getTotalBlockReplicatedSize(),
+            totalPendingAtDnSide);
+    return StorageCapacityDistributionResponse.newBuilder()
+        .setDataNodeUsage(nodeStorageReports)
+        .setGlobalStorage(storageMetrics)
+        .setGlobalNamespace(new GlobalNamespaceReport(namespaceMetrics.get("totalUsedNamespace"), 0))
+        .setUsedSpaceBreakDown(new UsedSpaceBreakDown(
+            namespaceMetrics.get("totalOpenKeySize"),
+            namespaceMetrics.get("totalCommittedSize"),
+            deletionPendingBytesByStage))
+        .build();
+  }
+
+  private List<DatanodeStorageReport> collectDatanodeReports() {
+    return nodeManager.getAllNodes().stream()
+        .map(this::getStorageReport)
+        .collect(Collectors.toList());
+  }
+
+  private long calculatePendingSizes() {
+    Map<String, Long> pendingDeletedDirSizes = new HashMap<>();
+    omdbInsightEndpoint.calculateTotalPendingDeletedDirSizes(pendingDeletedDirSizes);
+    Map<String, Long> pendingKeySize = new HashMap<>();
+    omdbInsightEndpoint.createKeysSummaryForDeletedKey(pendingKeySize);
+    long totalPendingDeletedDirSize = pendingDeletedDirSizes.getOrDefault("totalReplicatedDataSize", 0L);
+    long totalPendingKeySize = pendingKeySize.getOrDefault("totalReplicatedDataSize", 0L);
+    return totalPendingDeletedDirSize + totalPendingKeySize;
+  }
+
+  private long calculateOpenKeySizes() {
+    Map<String, Long> openKeySummary = new HashMap<>();
+    omdbInsightEndpoint.createKeysSummaryForOpenKey(openKeySummary);
+    omdbInsightEndpoint.createKeysSummaryForOpenMPUKey(openKeySummary);
+    Map<String, Long> pendingKeySize = new HashMap<>();
+    omdbInsightEndpoint.createKeysSummaryForDeletedKey(pendingKeySize);
```

Comment:
Calling `createKeysSummaryForDeletedKey()` in a method that should calculate open key sizes is incorrect